src/OpenApi/src/Comparers/OpenApiAnyComparer.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.OpenApi.Any;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal sealed class OpenApiAnyComparer : IEqualityComparer<IOpenApiAny>
+{
+    public static OpenApiAnyComparer Instance { get; } = new OpenApiAnyComparer();
+
+    public bool Equals(IOpenApiAny? x, IOpenApiAny? y)
```

Comment:
Should there be a shortcut for if the values are the same reference in these comparers, or are you intentionally computing the hash codes for that case?

Reply:
Nope -- using `ReferenceEquals` before computing the hash code is a good note here.